### ISSUE TITLE
HomeKit temperature adjustments apply until schedule, ignoring active override

### ISSUE THREAD

**Issue by PuzzledUser** [luc-ass/homebridge-evohome#149]
My system was set (EDIT: had an override active) to 19°C until 20:30. At 16:01, I used Apple Watch to increase the temperature to 20°C. However, the clever algorithm changed the validity time _until_ the next scheduled target change (which is 17°C @ 18:00). 

Why change the _until_ figure when setting the temperature? When using the Honeywell round thermostat to adjust the temperature it applies to any existing schedule or override.

Is it not possible just to send temp info to the server and leave the existing _until_ figure unchanged? If I hadn't spotted what was going on I'd have found it chilly later.

**Comment by PuzzledUser** [comment]
Yes, that's right @SeaneyC. To my mind, desired behaviour is that homebridge-evohome should respect your override time, not set a fresh override until the next scheduled target change time.

If you re-test in your scenario and use home bridge-evohome to change to 16°. You will find it sets that temperature until the next scheduled target change, not until your override time of 5pm.

The cure would be to not set a new override time, just set the desired target temperature.

**Comment by PuzzledUser** [comment]
I just tried hacking line 820:
```
//            .setHeatSetpoint(that.device.zoneID, value, nextScheduleTime)
            .setHeatSetpoint(that.device.zoneID, value, null) // try to stop override time
```

I passed null as the last parameter (which is done in all other places setHeatSetpoint is called). However, I discovered that this sets a permanent override. Not what I want.

I recall reading previously that it is tricky to _just_ set a new target temp. Hence the workaround of setting _until_ the next scheduled time. However, it must be possible to do this as Honeywell gear does it ok. So it's just figuring out what the API call is... Peesy. 😉

**Comment by PuzzledUser** [comment]
The behaviour is different for Honeywell Home and an adjustment of a rad valve in a Zone. For me a manual target adjustment is never propagated to the Evohome central unit, never reaches the server and hence never  gets to HomeKit via homebridge-evohome.

However, adjustments to a [Honeywell Round thermostat](https://www.resideo.com/gb/en/products/comfort/thermostats/smart-thermostats/y87rf2024-spdt-53a-24230-vac-rf-control-connectable-bdr91-y87rf2024/) quickly propagates its new target to the server and hence to HomeKit via home bridge-evohome.

### Bottom line
There is definitely an API to update just the target temp on the Honeywell online system at tccna.honeywell.com/WebAPI/ 

If we can find the call, we can solve my _problem_ and simply the existing code that calculates the next scheduled switch point (and cocks up across midnight).

**Comment by DenyTsjapanov** [comment]
Hi guys. Is there any development on this? The Honeywell app does not allow to delete ALL schedules. So it will always set at least 1 schedule. Alternatively you can set temperatures to "Permanent" in the app. That way, it ignores the schedules and leaves the temperature to set temperatures until changed manually. The Homebridge plugin doesn't allow that. Is there a way to achieve this still? Happy to help with the [device](https://www.resideo.com/nl/nl/producten/comfort/thermostaten/slimme-thermostaten/atp931gm4100-evohome-slimme-thermostaat-met-stuurmodule-atp931gm4100/) I have. Thanks in advance!

**Comment by luc-ass** [comment]
Fixed in `1.0.0-beta.1` (`npm install -g homebridge-evohome@beta`).

You were right that overwriting the end time was a decision the plugin had no
business making on its own. It is now a setting, `setpointMode`:

| Value | Behaviour |
| :-- | :-- |
| `keepExistingUntil` *(default)* | A running temporary override keeps its end time; only the temperature changes. With no override running, the change lasts until the next switchpoint. |
| `untilNextSwitchpoint` | Always until the next switchpoint — what 0.11.2 did. |
| `permanent` | Overrides the schedule until you pick "Auto" in HomeKit. |

The default is the behaviour you asked for: your 20 °C at 16:01 would now still
end at 20:30, not at 18:00.

@DenyTsjapanov, the third value covers your case — `"setpointMode": "permanent"`
sets the temperature until you change it back, the same as choosing Permanent in
the Honeywell app, without needing to delete any schedules. Turning a zone off
is always permanent regardless of the setting.

The log now also states which rule it applied and why, so it is visible rather
than guessable:
`Bathroom: target temperature 21 °C, until 18:30 UTC (end time of the running override)`.
